lib/web/auth/oauth2/index.js

```diff
@@ -50,18 +52,30 @@ function extractProfileAttribute (data, path) {
 }
 
 function parseProfile (data) {
+  const id = extractProfileAttribute(data, config.oauth2.userProfileIdAttr)
   const username = extractProfileAttribute(data, config.oauth2.userProfileUsernameAttr)
   const displayName = extractProfileAttribute(data, config.oauth2.userProfileDisplayNameAttr)
   const email = extractProfileAttribute(data, config.oauth2.userProfileEmailAttr)
 
   return {
-    id: username,
+    id: id || username,
     username: username,
     displayName: displayName,
     email: email
   }
 }
 
+function checkAuthorization (data, done) {
+  const roles = extractProfileAttribute(data, config.oauth2.rolesClaim)
+
+  if (config.oauth2.accessRole && roles) {
+    if (!roles.includes(config.oauth2.accessRole)) {
+      logger.error('oauth2 permission denied')
```

Comment:
I would adjust this error message a bit:

```suggestion
      logger.error('oauth2: user doesn't have the required role. Permission denied')
```

Without being aware of the quickest way to do that, I would also recommend to add the username into the server-side log in order to make debugging more pleasant.

Reply:
But in that case the username should be logged to the `debug`-output only. Don't put privacy-related information into a non-debug log.